proof/refine/ARM/SchedContextInv_R.thy

```diff
@@ -722,7 +722,7 @@ lemma refillResetRR_corres:
 lemma refillPopHead_corres:
   "corres (\<lambda>refill refill'. refill = refill_map refill')
               (pspace_aligned and pspace_distinct and sc_at sc_ptr and is_active_sc sc_ptr
-               and sc_refills_sc_at (\<lambda>refills. 1 < length refills) sc_ptr)
+               and (\<lambda>s. pred_map (\<lambda>cfg. Suc 0 < length (scrc_refills cfg)) (sc_refill_cfgs_of s) sc_ptr))
```

Comment:
I don't think this change is necessary. It would be better to adjust your use site to accommodate the current version, I'd think?

To give a further context, `refillPopHead_corres` is used in my `refillUnblockCheck_corres` proof, too. I should have thought about this, but there might be other overlapping things.
https://github.com/mktnk3/l4v/commit/d92215ab52ff919fa77149bd2490f9a25b86e9df

Reply:
I much prefer the `pred_map` formulations. But if you would like me to change this, I shall. We need a policy regarding `obj_at` versus `pred_map` alternatives

Reply:
well, isn't this actually the same thing as `is_active_sc`?

Reply:
I don't think we have this anywhere in the proofs. Carrying around that invariant would be its own pain

Reply:
Oh wait no. This wants length greater than 1, not 0. We can certainly have active scs which have length 1

Reply:
ah, yes, it's 1, not 0. So, it's the `is_active_sc` part that we could do without.

Reply:
I think, as a general rule, we'd better refrain from tweaking notations of existing lemmas if there is no semantic needs/improvements (i.e., you do need to fix that lemma). That is kind of a base line for multiple people working closely together on something.

As for the choice of `obj_at` or `pred_map`, we know that we can't easily agree on this point and there might not even be point in agreeing on one of those two choices right now. We are working towards converging (most likely to a third option) but it will be a while before that can happen.